### PR TITLE
Extension of the profile fields

### PR DESCRIPTION
**Pull request by LordSchirmer**
# Description

Extension and visual adjustments to the selection and display of profile fields.

New options:
- radio boxes
- check boxes,
- drop-down fields
- date fields

Fixes # (issue)

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# This PR has been tested in the following browsers:

- [X] Firefox

### CODE REVIEW

File: application/modules/user/views/profil/index.php [hunk 1]
```diff
@@ -136,30 +136,34 @@
...
-                        }
+            if (!$profileField->getShow()) {
+                continue;
+            }
+            $profileFieldName = $profileField->getKey();
+            $value = '';
+            foreach ($profileFieldsContent as $profileFieldContent) {
+                if ($profileFieldContent->getValue() && $profileField->getId() == $profileFieldContent->getFieldId()) {
+                    if ($profileField->getType() == 4) {
+                        $value = implode(', ', unserialize($profileFieldContent->getValue()));
```
Comment (blackcoder87) [inline application/modules/user/views/profil/index.php:147]: unserialize

File: application/modules/user/views/panel/profile.php [hunk 2]
```diff
@@ -96,52 +97,113 @@ class="form-control"
...
+                                        <?php if ($profileField->getShow() == 0) : ?>
+                                            <span class="input-group-addon check" rel="tooltip" title="<?=$this->getTrans('profileFieldHidden') ?>">
+                                                <span class="fa-solid fa-eye-slash"></span>
+                                            </span>
+                                        <?php endif; ?>
+                                    </div>
+                                <?php endforeach; ?>
+                            <!-- check -->
+                            <?php elseif ($profileField->getType() == 4) :
+                                $options = unserialize($profileField->getOptions());
```
Comment (blackcoder87) [inline application/modules/user/views/panel/profile.php:146]: unserialize

**Review by blackcoder87 — CHANGES_REQUESTED** [review]

**Comment by blackcoder87** [comment]
Might be unrelated to your changes, but I am seeing these errors:

index.php/admin/user/profilefields/treat/id/11
> Trying to access array offset on value of type null in application/modules/user/views/admin/profilefields/treat.php:88
> Trying to access array offset on value of type null in application/modules/user/views/admin/profilefields/treat.php:94

File: application/modules/user/models/ProfileField.php [hunk 3]
```diff
@@ -43,6 +43,13 @@ class ProfileField extends \Ilch\Model
      */
     protected $addition;
 
+    /**
+     * The options of the profile-field.
+     *
+     * @var text
```
[the remark this answers is not in the document]
Comment (LordSchirmer) [reply application/modules/user/models/ProfileField.php:49]: Yes is a string. I will change it.

File: application/modules/user/models/ProfileField.php [hunk 4]
```diff
@@ -179,6 +186,29 @@ public function setAddition($addition)
...
+        return $this->options;
+    }
+
+    /**
+     * Sets the options.
+     *
+     * @param text $options
+     * @return ProfileField
+     */
+    public function setOptions($options)
```
[the remark this answers is not in the document]
Comment (LordSchirmer) [reply application/modules/user/models/ProfileField.php:205]: Yes is a string. I will change it.

File: application/modules/user/views/admin/index/treatProfile.php [hunk 5]
```diff
@@ -50,15 +55,17 @@
...
         continue;
     }
-
     $profileFieldName = $profileField->getKey();
     $value = '';
     foreach ($profileFieldsContent as $profileFieldContent) {
         if ($profileFieldContent->getValue() && $profileField->getId() == $profileFieldContent->getFieldId()) {
-            $value = $profileFieldContent->getValue();
+            if ($profileField->getType() == 4) {
+                $value = implode(', ', unserialize($profileFieldContent->getValue()));
```
[the remark this answers is not in the document]
Comment (LordSchirmer) [reply application/modules/user/views/admin/index/treatProfile.php:63]: Ok I did not know. I will take a look.

File: application/modules/user/views/admin/profilefields/treat.php [hunk 6]
```diff
@@ -151,15 +250,38 @@ class="form-control"
...
+    var typeKey = $('#profileFieldType').find(':selected').val();
+    var typeDesc0 = '<?=$this->getTrans('profileFieldTypeDesc0') ?>';
+    var typeDesc1 = '<?=$this->getTrans('profileFieldTypeDesc1') ?>';
+    var typeDesc2 = '<?=$this->getTrans('profileFieldTypeDesc2') ?>';
+    var typeDesc3 = '<?=$this->getTrans('profileFieldTypeDesc3') ?>';
+    var typeDesc4 = '<?=$this->getTrans('profileFieldTypeDesc4') ?>';
+    var typeDesc5 = '<?=$this->getTrans('profileFieldTypeDesc5') ?>';
+    var typeDesc6 = '<?=$this->getTrans('profileFieldTypeDesc6') ?>';
+    var typeDesc7 = '<?=$this->getTrans('profileFieldTypeDesc7') ?>';
+    var iconArray = ['fa-regular fa-square','fa fa-list','fa-regular fa-face-smile','fa-regular fa-circle-check','fa-regular fa-square-check','far fa-caret-square-down','fa-regular fa-calendar-days'];
```
[the remark this answers is not in the document]
Comment (LordSchirmer) [reply application/modules/user/views/admin/profilefields/treat.php:282]: I will change

File: application/modules/user/views/admin/profilefields/treat.php [hunk 7]
```diff
@@ -91,40 +131,99 @@ class="form-control"
...
+
+    <!-- multi options -->
+    <?php $multiArr = [3, 4, 5]; ?>
+    <div class="profileFieldsMulti <?=(in_array($profileField->getType(), $multiArr)) ? '' : 'hidden' ?>">
+        <?php if ($profileField->getOptions()) { 
+            $options = json_decode($profileField->getOptions(), true);
+            $optionsAnz = 1; ?>
+            <div class="form-group">
+                <label for="profileFieldOptions" class="col-lg-2 control-label">
+                    <?=($optionsAnz == 1) ? $this->getTrans('profileFieldOptions') : ''; $optionsAnz++; ?>
```
Comment (blackcoder87) [inline application/modules/user/views/admin/profilefields/treat.php:190]: You initialize $optionsAnz with 1 and check here if it is 1. Can this ever have a different value than 1?
Doesn't look like it is inside a loop.

**Review by blackcoder87 — CHANGES_REQUESTED** [review]

File: application/modules/user/views/admin/profilefields/treat.php (hunk 7 above)
Comment (LordSchirmer) [reply application/modules/user/views/admin/profilefields/treat.php:190]: The goal was to only put the word "choice" once in front of the input fields. For me this was the only possibility in connection with the dynamic fields. Not nice but the goal reached :)

Comment (blackcoder87) [reply application/modules/user/views/admin/profilefields/treat.php]: I removed that workaround and don't see it being repeated. Can you confirm that?

Comment (LordSchirmer) [reply application/modules/user/views/admin/profilefields/treat.php]: The test showed no errors. Nice work! Thanks for your adjustments.

**Review by blackcoder87 — APPROVED** [review]

**Pull request merged**